### PR TITLE
Add basic runtime library

### LINKED ISSUE #23: Basic runtime library
For now, I think Cesium should have a runtime library in a separate assembly (written in C#). This should define nearly all of the C standard library facilities.

I have considered other approaches (write it in C with CLI interop and compile separately with Cesium, write all the code in headers), and consider the current idea the best for the development speed of this part.

This task will be considered closed after we add a runtime library with a couple of standard functions; no fancy stuff is required.

See also #22 (about a way to call the runtime library functions).

### PR DESCRIPTION
**Pull request by kant2002**
- puts
- strlen

Closes #23

### CODE REVIEW

**Comment by kant2002** [comment]
Errors seems to be from infra

File: Cesium.Runtime/Cesium.Runtime.csproj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (ForNeVeR) [inline Cesium.Runtime/Cesium.Runtime.csproj:4]: What do you think of targeting netstandard2.0 in our runtime library? I intend to support .NET Framework (not all versions, but at least the most recent ones) if it's not too tedious.

File: Cesium.IntegrationTests/stdio_test.c [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include <stdio.h>
+
+int main()
+{
+    puts("test");
```
Comment (ForNeVeR) [inline Cesium.IntegrationTests/stdio_test.c:5]: Probably this fails to parse due to string literals not being supported, yet.

File: Cesium.Runtime/Cesium.Runtime.csproj (hunk 1 above)
Comment (kant2002) [reply Cesium.Runtime/Cesium.Runtime.csproj:4]: Will change. I think that would be fine.

File: Cesium.IntegrationTests/stdio_test.c (hunk 2 above)
Comment (kant2002) [reply Cesium.IntegrationTests/stdio_test.c:5]: I will play. Maybe I will change test somehow

File: Cesium.Runtime/Cesium.Runtime.csproj (hunk 1 above)
Comment (kant2002) [reply Cesium.Runtime/Cesium.Runtime.csproj]: If we insist on having char as UTF-8 char, then for netstandard2.0 I cannot use `Marshal.PtrToStringUTF8` and we have to manually convert string to UTF8.

File: Cesium.IntegrationTests/stdio_test.c (hunk 2 above)
Comment (kant2002) [reply Cesium.IntegrationTests/stdio_test.c:5]: There nothing which I can do. I try following variants with same error in compiler, so I probably have to remove this file from integration tests.
```
//char t[5] = { 't', 'e', 's', 't', '\0', };
char t[5] = { 0x74, 0x65, 0x73, 0x74, 0, };
puts(t);
//puts("test");
```

File: Cesium.Runtime/Cesium.Runtime.csproj (hunk 1 above)
Comment (ForNeVeR) [reply Cesium.Runtime/Cesium.Runtime.csproj]: Hm, alright. For the time being, let's live with `net6.0`, then.

File: Cesium.IntegrationTests/stdio_test.c (hunk 2 above)
Comment (ForNeVeR) [reply Cesium.IntegrationTests/stdio_test.c:5]: All of these aren't supported, yet :)

I propose to put this aside for now (a couple of days max; probably a dozen of hours) and give me some time to implement at least `const char* val = "hello world";`

Comment (kant2002) [reply Cesium.IntegrationTests/stdio_test.c:5]: I think if I can add some function which does not rely on char, maybe math functions and test these parts of runtime lib. That way we will move forward.

File: Cesium.Runtime/Cesium.Runtime.csproj (hunk 1 above)
Comment (kant2002) [reply Cesium.Runtime/Cesium.Runtime.csproj]: I have both `netstandard2.0` version and `net6.0` version. Should I remove `netstandard2.0` ?

Comment (ForNeVeR) [reply Cesium.Runtime/Cesium.Runtime.csproj]: I see, you've implemented the method yourself.

Let's leave this in the current state, then. `netstandard2.0;net6.0` looks okay to me for now.

File: Cesium.IntegrationTests/stdio_test.c (hunk 2 above)
Comment (ForNeVeR) [reply Cesium.IntegrationTests/stdio_test.c:5]: That would be okay, yeah. Feel free to omit this particular integration test for now.

**Comment by kant2002** [comment]
I have no idea why this is green. This is consistently fails on my Linux (Gentoo) machine.

**Review by ForNeVeR — APPROVED** [review]
This is awesome!

**Pull request merged**